kafka/src/main/java/io/specmesh/kafka/provision/schema/SchemaReaders.java

```diff
@@ -111,7 +111,12 @@ private static AvroSchema toAvroSchema(final DetectedSchema schema) {
 
             final Map<String, String> resolvedReferences =
                     schema.references().stream()
-                            .collect(toMap(DetectedSchema::subject, DetectedSchema::content));
+                            .collect(
+                                    toMap(
+                                            DetectedSchema::subject,
+                                            DetectedSchema::content,
+                                            (s1, s2) -> s1,
+                                            LinkedHashMap::new));
```

Comment:
Using a linked hashmap here fixes #427.

Tested by `other.domain.Common` now referencing `other.domain.CommonOther`. Which leads to there being multiple entries in the map and the order being important.

`AvroSchema` has a badly designed API!